frontend/src/employee-mobile-frontend/components/attendances/actions/MarkAbsentBeforehand.tsx

```diff
@@ -74,6 +86,12 @@ export default React.memo(function MarkAbsentBeforehand() {
     loadFutureAbsences(childId)
   }, [loadFutureAbsences, childId])
 
+  useEffect(() => {
+    if (absences.isSuccess && absences.value.length > 1) {
+      setAbsenceRanges(groupAbsencesByDateRange(absences))
+    }
+  }, [absences])
+
```

Comment:
In general I don't think storing `absenceRanges` in a separate `useState` hook is a good idea. In reality it's just data computed from `absences` and could just be stored in a variable. If the idea is to minimize unnecessary computations, use `useMemo` instead.